ci/src/main/java/ci/ContinuousIntegrationServer.java

```diff
@@ -3,30 +3,60 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
 
 import java.io.IOException;
 
 import org.eclipse.jetty.server.Server;
 import org.eclipse.jetty.server.Request;
 import org.eclipse.jetty.server.handler.AbstractHandler;
 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.InputStream;
+
 /**
  Skeleton of a ContinuousIntegrationServer which acts as webhook
  See the Jetty documentation for API documentation of those classes.
  */
-public class ContinuousIntegrationServer extends AbstractHandler
-{
+public class ContinuousIntegrationServer extends AbstractHandler {
     public void handle(String target,
                        Request baseRequest,
                        HttpServletRequest request,
                        HttpServletResponse response)
-            throws IOException, ServletException
-    {
+            throws IOException, ServletException {
+        String githubEvent = request.getHeader("X-Github-Event");
+
+        if(!githubEvent.equals("")){
+            switch(githubEvent){
+                case "push":
+                	//Isolate the branchName from the payload 
+                    String payload = getRequestPayload(request);
+                    String issue = "issue%2F";
```

Comment:
is there no function to get the branch name? like on line 28, where you get the type of event using the getHeader function.

Reply:
One could parse the payload that does include it. I did not know what information was necesarry so I just did some general functions.

Reply:
also, doesn't github send a json object? I thought parsing and extracting information would be a lot easier.